lib/test_prof/recipes/rspec/let_it_be.rb

```diff
@@ -145,7 +191,22 @@ def define_let_it_be_methods(identifier, **modifiers)
       end
       record
     end
+
+    config.register_modifier :freeze do |record, val, mods|
+      next record if val == false
```

Comment:
Shouldn't we add the record to the `stoplist` here?

And make `stoplist` a _global registry_ of `let_it_be` objects. How it could be implemented? I'm still thinking about WeakMap.

That could solve the edge cases you mentioned in https://github.com/palkan/test-prof/pull/178#pullrequestreview-367996478.

Reply:
It seems that it could be possible to detect the moment where we open the outermost transaction, i.e. where `within_before_all?` flips from `false` to `true`. Then it makes it possible to reset the set of `let_it_be`d objects when the older entries don't apply anymore.

That would allow reducing the lifecycle of this set from suite to one spec (or even an example group if this example group does not have common `let_it_be`/`before_all` with other example groups).